openauth-template/src/index.ts

```diff
@@ -0,0 +1,94 @@
+import { authorizer, createSubjects } from "@openauthjs/openauth";
+import { CloudflareStorage } from "@openauthjs/openauth/storage/cloudflare";
+import { PasswordAdapter } from "@openauthjs/openauth/adapter/password";
+import { PasswordUI } from "@openauthjs/openauth/ui/password";
+import { object, string } from "valibot";
+
+// This value should be shared between the OpenAuth server Worker and other
+// client Workers that you connect to it, so the types and schema validation are
+// consistent.
+const subjects = createSubjects({
+  user: object({
+    id: string(),
+  }),
+});
+
+export default {
+  async fetch(request: Request, env: Env, ctx: ExecutionContext) {
+    // This top section is just for demo purposes. In a real setup another
+    // application would redirect the user to this Worker to be authenticated,
+    // and after signing in or registering the user would be redirected back to
+    // the application they came from. In our demo setup there is no other
+    // application, so this Worker needs to do the initial redirect and handle
+    // the callback redirect on completion.
+    const url = new URL(request.url);
+    if (url.pathname === "/") {
+      return Response.redirect(
+        url.origin +
+          `/authorize?client_id=your-client-id&redirect_uri=${encodeURIComponent(url.origin + "/callback")}&response_type=code`,
```

Comment:
Should the client ID be an env var?

Reply:
No, since client ID represents some other application that redirects the user to this Worker to authenticate. This section of the template is just a stub for demo purposes, and you wouldn't want to customize this.